solidity/contracts/hooks/ArbL2ToL1Hook.sol

```diff
@@ -50,30 +52,41 @@ contract ArbL2ToL1Hook is AbstractMessageIdAuthHook {
         uint32 _destinationDomain,
         bytes32 _ism,
         address _arbSys,
-        uint256 _gasQuote
+        address _childHook
     ) AbstractMessageIdAuthHook(_mailbox, _destinationDomain, _ism) {
         arbSys = ArbSys(_arbSys);
-        GAS_QUOTE = _gasQuote;
+        childHook = AbstractPostDispatchHook(_childHook);
     }
 
+    /// @inheritdoc IPostDispatchHook
     function hookType() external pure override returns (uint8) {
         return uint8(IPostDispatchHook.Types.ARB_L2_TO_L1);
     }
 
+    /// @inheritdoc AbstractPostDispatchHook
     function _quoteDispatch(
-        bytes calldata,
-        bytes calldata
+        bytes calldata metadata,
+        bytes calldata message
     ) internal view override returns (uint256) {
-        return GAS_QUOTE;
+        return
+            metadata.msgValue(0) + childHook.quoteDispatch(metadata, message);
```

Comment:
realizing that the IGP adds the default ISM overhead which this message would not incur?

Reply:
Yes, we'll need to override that when specifying metadata to this hook. Maybe I can add ARB_ISM_GAS_USAGE as a constant to the hook which overrides the metadata? This is also an argument for specifying the childHook ie igp and not the abstract childHook.

Reply:
I dont think we can override the IGP gas limit

Reply:
https://github.com/hyperlane-xyz/hyperlane-monorepo/blob/2cc51c1640db7757a08ecd16241a49efd17aff3c/solidity/contracts/hooks/igp/InterchainGasPaymaster.sol#L272